src/app/shared/checker-workflow.service.ts

```diff
@@ -0,0 +1,163 @@
+            } else {
+                this.workflowsService.getWorkflow(id).subscribe((workflow: Workflow) => {
+                    this.checkerWorkflow$.next(workflow);
+                    if (this.router.url.includes('workflows')) {
+                        this.workflowService.upsertWorkflowToWorkflow(workflow);
+                    }
+                }, error => this.checkerWorkflow$.next(null));
+            }
+        });
+    }
+
+    /**
+     * Go to checker workflow
+     */
+    public goToCheckerWorkflow(): void {
+        const workflow = this.checkerWorkflow;
+        if (workflow) {
+            this.goToEntry(this.publicPage, workflow.full_workflow_path, 'workflow');
+        } else {
+            console.log('This entry has no checker workflow.');
+        }
+    }
+
+    /**
+     * Go to parent entry (could be a tool or workflow)
+     */
+    public goToParentEntry(): void {
+        const parentId = (<Workflow>this.entry).parent_id;
+        if (!parentId) {
+            console.log('This entry is not a checker workflow and has no parent entry.');
+            return;
+        }
+        if (this.publicPage) {
+            this.workflowsService.getPublishedWorkflow(parentId).first().subscribe((workflow: Workflow) => {
+                this.goToEntry(this.publicPage, workflow.full_workflow_path, 'workflow');
+            }, error => this.containersService.getPublishedContainer(parentId).first().subscribe((tool: DockstoreTool) => {
+                this.goToEntry(this.publicPage, tool.tool_path, 'tool');
+            }, error2 => console.log('Can not get parent entry')));
+        } else {
+            this.workflowsService.getWorkflow(parentId).first().subscribe((workflow: Workflow) => {
+                this.goToEntry(this.publicPage, workflow.full_workflow_path, 'workflow');
+            }, error => this.containersService.getContainer(parentId).first().subscribe((tool: DockstoreTool) => {
+                this.goToEntry(this.publicPage, tool.tool_path, 'tool');
+            }, error2 => console.log('Can not get parent entry')));
+        }
+    }
+
+    /**
+     * Go to entry
+     * @param path Full path of entry
+     * @param entryType Either 'tool' or 'workflow'
+     */
+    public goToEntry(publicPage: boolean, path: string, entryType: 'tool' | 'workflow'): void {
+        let url: string;
+        if (publicPage) {
+            url = '/' + entryType + 's/' + path;
+        } else {
+            url = '/my-' + entryType + 's/' + path;
+        }
+        this.router.navigateByUrl(url);
+    }
+
+    /**
+     * Determines if a given entry is a workflow or not (tool).
+     * Returns true if entry is a workflow, returns false if entry is not a workflow (i.e. a tool)
+     * @param entry The entry to check
+     */
+    private isEntryAWorkflow(entry: Entry): boolean {
+        if (!entry) {
+            return null;
+        }
+        if (entry.hasOwnProperty('is_checker')) {
```

Comment:
Minor nitpick: shouldn't these five lines just be 
`return entry.hasOwnProperty('is_checker')`?